### ISSUE TITLE
`(Segmentation fault): protoc failed` when building

### ISSUE THREAD

**Issue by peleyal** [protocolbuffers/protobuf#9172]
Disclaimer: I just upgraded my MAC to Monterey (12.0.1) a few days ago, and **I think** that this issue started just after.

**What version of protobuf and what language are you using?**
Version: v3.13.0 and I also tried v3.19.1
Language: Python

**What operating system (Linux, Windows, ...) and version?**
MacOS Monterey (12.0.1) 

**What runtime / compiler are you using (e.g., python version or gcc version)**
Python 3.9.6

**What did you do?**
(I Just pushed https://github.com/peleyal/peleyal/commit/5dc65aeed72b5b6c1a14847ec10f344c7dc33166 for that purpose...)
1. Clone the stupid repo https://github.com/peleyal/peleyal
2. bazel build ...

**What did you expect to see**
Success build :)

**What did you see instead?**

```
peleyal$ bazel build ...
INFO: SHA256 (https://golang.org/dl/?mode=json&include=all) = d8c610666c9794bb761bc0dacd7ebc9edae139a4b6044cfee291a5623136317c
INFO: Analyzed 3 targets (25 packages loaded, 831 targets configured).
INFO: Found 3 targets...
INFO: From Linking external/com_google_protobuf/libprotobuf.a:
/Library/Developer/CommandLineTools/usr/bin/libtool: file: bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/_objs/protobuf/error_listener.o has no symbols
INFO: From Linking external/com_google_protobuf/libprotobuf_lite.a:
/Library/Developer/CommandLineTools/usr/bin/libtool: file: bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/_objs/protobuf_lite/io_win32.o has no symbols
INFO: From Linking external/com_google_protobuf/libprotobuf_lite.a [for host]:
/Library/Developer/CommandLineTools/usr/bin/libtool: file: bazel-out/host/bin/external/com_google_protobuf/_objs/protobuf_lite/io_win32.o has no symbols
ERROR: /private/var/tmp/_bazel_peleyal/030f6eee7f9fec574720f87e326c8c2a/external/com_google_protobuf/BUILD:971:17: ProtoCompile external/com_google_protobuf/python/google/protobuf/compiler/plugin_pb2.py failed: (Segmentation fault): protoc failed: error executing command bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/protoc '--python_out=bazel-out/darwin-fastbuild/bin/external/com_google_protobuf/python' -Iexternal/com_google_protobuf/python ... (remaining 2 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
INFO: Elapsed time: 168.081s, Critical Path: 20.27s
INFO: 410 processes: 35 internal, 375 darwin-sandbox.
FAILED: Build did NOT complete successfully
```

Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

**Anything else we should know about your project / environment**

Thank you! I really appreciate your time.
Eyal

**Comment by acozzette** [comment]
I would guess that this is probably the same issue as #8884.

**Comment by jvolkman** [comment]
This is fixed in Bazel 4.2.2 by https://github.com/bazelbuild/bazel/pull/14275

**Comment by peleyal** [comment]
Thanks! It is indeed working now!!! 🥇 🥇 🥇 🥇 🥇
